lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -159,6 +159,7 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
 
     if (killAfterTimeoutThread) {
+      usleep(1000); // Give the thread some time to actually reach wait_for()
```

Comment:
This isn't safe... `usleep` is never a safe way to resolve a race since the OS may schedule things in such a way it doesn't have the intended effect.